examples/ChatQnA/deploy/k8s_helm.md

```diff
@@ -0,0 +1,497 @@
+curl http://localhost:8888/v1/chatqna -H "Content-Type: application/json" -d '{
+     "model": "Intel/neural-chat-7b-v3-3",
+     "messages": "What is the revenue of Nike in 2023?"
+     }'
+
+```
+Here is the output for your reference:
+```bash
+data: b' O', data: b'PE', data: b'A', data: b' stands', data: b' for', data: b' Organization', data: b' of', data: b' Public', data: b' Em', data: b'ploy', data: b'ees', data: b' of', data: b' Alabama', data: b'.', data: b' It', data: b' is', data: b' a', data: b' labor', data: b' union', data: b' representing', data: b' public', data: b' employees', data: b' in', data: b' the', data: b' state', data: b' of', data: b' Alabama', data: b',', data: b' working', data: b' to', data: b' protect', data: b' their', data: b' rights', data: b' and', data: b' interests', data: b'.', data: b'', data: b'', data: [DONE]
+```
+which is essentially the following sentence:
+```
+OPEA stands for Organization of Public Employees of Alabama. It is a labor union representing public employees in the state of Alabama, working to protect their rights and interests.
+```
+In the upcoming sections we will see how this answer can be improved with RAG.
+
+### Dataprep Microservice
+Use the following command to forward traffic from your local machine to the service running in the Kubernetes cluster:
+```bash
+kubectl port-forward svc/chatqna-data-prep 6007:6007
+```
+Follow the below steps in a different terminal.
+
+If you want to add/update the default knowledge base, you can use the following
+commands. The dataprep microservice extracts the texts from variety of data
+sources, chunks the data, embeds each chunk using embedding microservice and
+store the embedded vectors in the redis vector database.
+
+this example leverages the OPEA document for its RAG based content. You can download the [OPEA document](https://opea-project.github.io/latest/_downloads/41c91aec1d47f20ca22350daa8c2cadc/what_is_opea.pdf) and upload it using the UI.
+
+
+Local File `what_is_opea.pdf` Upload:
+
+```
+curl -X POST "http://localhost:6007/v1/dataprep" \
+     -H "Content-Type: multipart/form-data" \
+     -F "files=@./what_is_opea.pdf"
+```
+
+This command updates a knowledge base by uploading a local file for processing.
+Update the file path according to your environment.
+
+You should see the following output after successful execution:
+```
+{"status":200,"message":"Data preparation succeeded"}
+```
+For advanced usage of the dataprep microservice refer [here](#dataprep-microservice-%28advanced%29)
+
+### MegaService After RAG Dataprep
+
+Use the following command to forward traffic from your local machine to the service running in the Kubernetes cluster:
+```bash
+kubectl port-forward svc/chatqna 8888:8888
+```
+Similarly, follow the below steps in a different terminal.
+
+```
+curl http://localhost:8888/v1/chatqna -H "Content-Type: application/json" -d '{
+     "model": "Intel/neural-chat-7b-v3-3",
+     "messages": "What is OPEA?"
+     }'
+
```

Comment:
Please remove these redundant empty lines from all the command examples.  They do not look good.